src/ompl/util/RandomNumbers.h

```diff
@@ -43,8 +43,18 @@
 #include <boost/random/variate_generator.hpp>
 #include <cassert>
 
+#include <ompl/util/ClassForward.h>
+
+//The basic Eigen include
+#include <Eigen/Core>
```

Comment:
Forgive me if I'm misunderstanding your issue here, but I don't think Mark was implying to forward declare the Eigen structures; he was implying putting all of ProlateHyperSpheroid's private members inside a PIMPL object that is only defined in the cpp file. That way, Eigen types are not even referred to in RandomNumbers.h at all. [Here's an example.](http://c2.com/cgi/wiki?PimplIdiom)

Like I said, I apologize if you've already tried this and are experiencing issues; if this is the case, give us more details about the problem and we can work through it.